### PR TITLE
GitHub Actions

### PR DESCRIPTION
**Pull request by JesperIRL**
Encouraging developers to enable GHA on their personal forks.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Reviewers
 * [Kevin Rushforth](https://openjdk.java.net/census#kcr) (@kevinrushforth - no project role) ⚠️ Review applies to [69ae1fa4](https://git.openjdk.java.net/guide/pull/80/files/69ae1fa43a2ed0eab548dff8a2d9cd236730a947)
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/guide pull/80/head:pull/80` \
`$ git checkout pull/80`

Update a local copy of the PR: \
`$ git checkout pull/80` \
`$ git pull https://git.openjdk.java.net/guide pull/80/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 80`

View PR using the GUI difftool: \
`$ git pr show -t 80`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/guide/pull/80.diff">https://git.openjdk.java.net/guide/pull/80.diff</a>

</details>

### CODE REVIEW

File: src/index.md [hunk 1]
```diff
@@ -1298,6 +1298,14 @@ The fix for the main issue should remove the test from the ProblemList or remove
 
 After a failure is handled by excluding a test, the main JBS issue should be re-triaged and possibly given a new priority. This should be handled by the standard triage process. A test exclusion results in an outage in our testing. This outage should be taken into consideration when triaging, in addition to the impact of the bug itself.
 
+## GitHub Actions
+
+GitHub has a feature called **GitHub Actions** (GHA) that can be used to automate testing. The GHA is executed whenever a PR is created or updated in the repository. The GHA in the JDK project is configured to run a set of tests that is commonly known as **tier 1**. This is a relatively fast, small set of tests that tries to verify that your change didn't break the JDK completely. In tier 1 the JDK is built on a small set of platforms including (but not necessarily limited to) Linux, Windows, and MacOS, and a few tests are executed using these builds.
```
Comment (kevinrushforth) [inline src/index.md:1303]: One correction: Once enabled in your repo it runs on a push to a branch in your repo, not just when you create a PR. Skara will show the results of the test run when you create or update a PR.

Comment (JesperIRL) [reply src/index.md]: Thanks for this clarification!

**Review by kevinrushforth — APPROVED** [review]
Looks good.

Magnus has a fair point about GHA runs being prone to things like external updates (although there are usually folks in the community who will jump on those and get our GHA scripts fixed pretty quickly when needed, they do cause spurious errors for a while when it happens). I think his recommendation to try to reproduce failures on an environment you control is a good one.

**Comment by JesperIRL** [comment]
Thanks for these comments. I have added a few sentences to highlight this.

File: src/index.md [hunk 2]
```diff
@@ -1304,6 +1304,8 @@ GitHub has a feature called **GitHub Actions** (GHA) that can be used to automat
 
 In addition to the testing you run manually before publishing your changes, it's recommended that you take advantage of this automated testing that the GHA offers. To enable this on your personal fork of the JDK on GitHub go to the "Actions" tab and click the big green button saying "I understand my workflows, go ahead and enable them". If you don't understand these workflows there's a link to the actual file that describes them right below the green button.
 
+In case of failures in the GHA it's always a good start to try to reproduce the failure locally on a machine where you have better control and easier access to a debug environment. There has been cases where the GHA has failed due to issues unrelated to the change being tested, e.g. because the GHA environment was updated and changes was needed to the JDK GHA configuration. The configuration is in general updated fairly quickly, so in cases were you can't reproduce the failure locally, consider re-running the GHA later.
```
Comment (kevinrushforth) [inline src/index.md:1307]: Minor: that should be "There _have_ been cases where..." and later "and changes _were_ needed to the JDK GHA"

**Review by kevinrushforth — APPROVED** [review]
The new sentence look good to me with two wording comments inline.

File: src/index.md (hunk 2 above)
Comment (JesperIRL) [reply src/index.md]: Good catch! Fixed.

**Review by kevinrushforth — APPROVED** [review]

**Comment by prrace** [comment]
1) Conversely to Magnus' point it might build with GHA but not with OpenJDK recommended tool chains.
Still this is STREETS ahead of you not building at all on Windows (say) since you don't have access.

2) You could say that you can get *some* idea of what tier1 means for hotspot and libraries by looking at  the jtreg TEST.groups files in the repo (you can provide links) and  you'll quickly see that many important tests and whole areas are not  covered at all by tier1.

**Comment by JesperIRL** [comment]
Yes, I think you have a good point here @prrace. I've added a note about running platforms one may not otherwise have access to and mentioned the TEST.groups files with a link.

**Review by irisclark — APPROVED** [review]

**Pull request closed without merging**